config/v1/types_infrastructure.go

```diff
@@ -500,6 +506,43 @@ type AWSResourceTag struct {
 	Value string `json:"value"`
 }
 
+// DNSAndLBConfiguration contains information about API and API-Int Load Balancers
+// and who is responsible for that configuration.
+type DNSAndLBConfiguration struct {
+	// When the user has pre-created the Load Balancers for API and API-Int and the
+	// the DNS solution has already been configured, this field is set to "UserConfig".
+	// When the cluster is resposible for its own DNS configuration and LB creation,
+	// this field is set to "Default".
+	// +default="Default"
+	// +kubebuilder:default:="Default"
+	// +kubebuilder:validation:Enum="UserConfig";"Default"
+	// +unionDiscriminator
+	// +optional
+	Provider string `json:"provider,omitempty"`
```

Comment:
Provider should be required
```suggestion
	// +kubebuilder:validation:Required
	Provider string `json:"provider"`
```
You should also use a type alias with constants for the different options for the enum

Reply:
How about "userProvided" and "clusterProvided"?

Reply:
Yeah lets go with that for now